### PR TITLE
Streamline the use of `is_empty` statements.

### PR DESCRIPTION
**Pull request by MathieuDutSik**
## Motivation

The `fn size` question is implemented for the batch. But the more common idiom is `fn len`. This correction led to a number of changes.

## Proposal

The following changes are done:
* The `fn size` is changed into `fn len`.
* The `is_empty` is introduced for the `Batch` type.
* It is used in the `server.rs` of the `linera-storage-service`.
* The early termination `if batch.is_empty()` is added to each of the `write_batch` functions.

The test for the batch being empty is essentially free but the gain in speed if matching is significant. Therefore, I find it to be a good idea to leave possibly redundant calls.

## Test Plan

CI.

## Release Plan

Not much important to the TestNet / DevNet.

## Links

None.

### CODE REVIEW

File: linera-views/src/backends/dual.rs [hunk 1]
```diff
@@ -199,6 +199,9 @@ where
     const MAX_VALUE_SIZE: usize = usize::MAX;
 
     async fn write_batch(&self, batch: Batch) -> Result<(), Self::Error> {
+        if batch.is_empty() {
+            return Ok(());
+        }
```
Comment (ma2bd) [inline linera-views/src/backends/dual.rs:204]: I wouldn't add this one and keep `dual.rs` unopinionated (e.g. concretely doesn't affect metrics).

Comment (ma2bd) [reply linera-views/src/backends/dual.rs:204]: In general, how often are batches empty?

Comment (MathieuDutSik) [reply linera-views/src/backends/dual.rs:204]: Quite often when I did some test some time ago. But not recently.
Ok, for `dual.rs`.

File: linera-views/src/batch.rs [hunk 2]
```diff
@@ -186,7 +186,7 @@ impl Batch {
     }
 
     /// The total size of the batch
-    pub fn size(&self) -> usize {
+    pub fn len(&self) -> usize {
```
Comment (ma2bd) [inline linera-views/src/batch.rs:189]: I understand why you'd like to imitate `Vec::len` but we're already using "batch size" everywhere and I believe it's totally fine.

File: linera-views-derive/src/lib.rs [hunk 3]
```diff
@@ -263,7 +263,9 @@ fn generate_save_delete_view_code(input: ItemStruct) -> TokenStream2 {
                 #increment_counter
                 let mut batch = Batch::new();
                 #(#flushes)*
-                self.context().write_batch(batch).await?;
+                if !batch.is_empty() {
```
Comment (ma2bd) [inline linera-views-derive/src/lib.rs:266]: nice!

File: linera-views/src/backends/metering.rs [hunk 4]
```diff
@@ -420,7 +420,7 @@ where
         self.counter
             .write_batch_size
             .with_label_values(&[])
-            .observe(batch.size() as f64);
+            .observe(batch.len() as f64);
```
Comment (ma2bd) [inline linera-views/src/backends/metering.rs:423]: This doesn't match `write_batch_size` any more

File: linera-views/src/batch.rs (hunk 2 above)
Comment (MathieuDutSik) [reply linera-views/src/batch.rs:189]: Ok, so let's revert that.

File: .github/workflows/rust.yml [hunk 5]
```diff
@@ -140,7 +140,7 @@ jobs:
 
   benchmark-test:
     runs-on: ubuntu-latest
-    timeout-minutes: 20
+    timeout-minutes: 30
```
Comment (ma2bd) [inline .github/workflows/rust.yml:143]: This seems unrelated to this PR

**Review by ma2bd — APPROVED** [review]
